### PR TITLE
ROX-12320: Update the list of tests that should be considered "version compatibility"

### PR DESCRIPTION
**Pull request by jschnath**
## Description

Created a new -DGroup COMPATIBILITY, called that group within compatibility-tests, marked groovy tests as compatibility.

## Checklist
- [ ] Investigated and inspected CI test results
~- [ ] Unit test and regression tests added~
~- [ ] Evaluated and added CHANGELOG entry if required~
~- [ ] Determined and documented upgrade steps~
~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

## Testing Performed

Still need to manually run those tests and refine the list.

### CODE REVIEW

**Review by Maddosaurus — APPROVED** [review]
That looks like a nice change!
Some questions I've got: 
* Do we have some documentation/decision helper for the different QA test groups? Imagine a future dev writing new tests. When should they add their tests to the Compatibility group?
* Out of Curiosity: How does Prow/the CI Bot pick up this new group? I think I saw you trying to test it in the PR, right?

**Comment by fredrb** [comment]
Doesn't seem like the tests are actually running: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/stackrox_stackrox/3169/pull-ci-stackrox-stackrox-master-gke-version-compatibility-tests/1572890990530793472/build-log.txt 😞

File: qa-tests-backend/src/test/groovy/UpgradesTest.groovy [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import com.google.protobuf.util.JsonFormat
+import groups.COMPATIBILITY
```
Comment (connorgorman) [inline qa-tests-backend/src/test/groovy/UpgradesTest.groovy:2]: this isn't used here

File: qa-tests-backend/src/test/groovy/PolicyConfigurationTest.groovy [hunk 2]
```diff
@@ -14,6 +14,7 @@ import io.stackrox.proto.storage.ScopeOuterClass.Scope
 import common.Constants
 import groups.BAT
 import groups.SMOKE
+import groups.COMPATIBILITY
```
Comment (connorgorman) [inline qa-tests-backend/src/test/groovy/PolicyConfigurationTest.groovy:17]: this isn't used?

File: qa-tests-backend/src/test/groovy/AutocompleteTest.groovy [hunk 3]
```diff
@@ -39,7 +40,7 @@ class AutocompleteTest extends BaseSpecification {
     }
 
     @Unroll
-    @Category([BAT])
+    @Category([BAT, COMPATIBILITY])
```
[the remark this answers is not in the document]
Comment (jschnath) [reply qa-tests-backend/src/test/groovy/AutocompleteTest.groovy:43]: My thought process is that API calls would be the most affected by version mismatches, but if this API doesn't happen from sensor it makes sense to leave the test out.

File: qa-tests-backend/src/test/groovy/UpgradesTest.groovy (hunk 1 above)
Comment (jschnath) [reply qa-tests-backend/src/test/groovy/UpgradesTest.groovy:2]: Good catches, I had added tests from these two files to COMPATIBILITY then removed them again later.

File: qa-tests-backend/src/test/groovy/PolicyConfigurationTest.groovy (hunk 2 above)
Comment (jschnath) [reply qa-tests-backend/src/test/groovy/PolicyConfigurationTest.groovy:17]: Good catches, I had added tests from these two files to COMPATIBILITY then removed them again later.

File: qa-tests-backend/src/test/groovy/DefaultPoliciesTest.groovy [hunk 4]
```diff
@@ -225,7 +226,7 @@ class DefaultPoliciesTest extends BaseSpecification {
         waitForViolation(K8S_DASHBOARD,  "Kubernetes Dashboard Deployed", 30)
     }
 
-    @Category(BAT)
+    @Category(BAT, COMPATIBILITY)
```
[the remark this answers is not in the document]
Comment (jschnath) [reply qa-tests-backend/src/test/groovy/DefaultPoliciesTest.groovy:229]: I went through it again, let's leave that one out then. Better to keep the list of COMPATIBILITY tests shorter.

File: qa-tests-backend/src/test/groovy/AutocompleteTest.groovy (hunk 3 above)
[the remark this answers is not in the document]
Comment (jschnath) [reply qa-tests-backend/src/test/groovy/AutocompleteTest.groovy:43]: I think I prefer to leave this one in for now, it's a relatively simple test too.

**Pull request merged**